internal/loader/configuration_client_manager.go

```diff
@@ -0,0 +1,478 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+package loader
+
+import (
+	"azappconfig/provider/internal/properties"
+	"context"
+	"fmt"
+	"math"
+	"net"
+	"net/url"
+	"strconv"
+	"strings"
+	"time"
+
+	acpv1 "azappconfig/provider/api/v1"
+
+	"math/rand"
+
+	"github.com/Azure/azure-sdk-for-go/sdk/azcore"
+	"github.com/Azure/azure-sdk-for-go/sdk/azcore/policy"
+	"github.com/Azure/azure-sdk-for-go/sdk/azidentity"
+	"github.com/Azure/azure-sdk-for-go/sdk/data/azappconfig"
+	"github.com/google/uuid"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+)
+
+//go:generate mockgen -destination=mocks/mock_configuration_client_manager.go -package mocks . ClientManager
+
+type ConfigurationClientManager struct {
+	ReplicaDiscoveryEnabled   bool
+	SpecifiedClient           *azappconfig.Client
+	DynamicClients            []*ConfigurationClientWrapper
+	ValidDomain               string
+	Endpoint                  string
+	ClientsFallbackStatus     map[string]*ConfigurationClientWrapper
```

Comment:
Why we need this?  Could use DynamicClients   and StaticClients directly

Reply:
Do you mean we don't need to update dynamicClients fallback status when autofailover enabled?

Reply:
I'm not saying we don't need to update fallback of dynamicClients, I mean why we don't update it in `DynamicClientwrappers` directly, not need to use another object which is only for managing the fallbacks

Reply:
updated